service_notification/src/main/kotlin/org/thechance/service_notification/domain/usecases/ITopicManagementUseCase.kt

```diff
@@ -0,0 +1,44 @@
+package org.thechance.service_notification.domain.usecases
+
+import org.koin.core.annotation.Single
+import org.thechance.service_notification.domain.entity.InternalServerErrorException
+import org.thechance.service_notification.domain.gateway.IDatabaseGateway
+import org.thechance.service_notification.domain.gateway.IPushNotificationGateway
+import org.thechance.service_notification.endpoints.ALREADY_EXISTS_TOPIC
+import org.thechance.service_notification.endpoints.TOPIC_NOT_EXISTS
+
+interface ITopicManagementUseCase {
+    suspend fun createTopic(topic: String): Boolean
+
+    suspend fun getTopics(): List<String>
+
+    suspend fun subscribeToTopic(topic: String, token: String): Boolean
+
+    suspend fun unsubscribeFromTopic(topic: String, token: String): Boolean
+}
+
+@Single
+class TopicManagementUseCase(
+    private val pushNotificationGateway: IPushNotificationGateway,
+    private val databaseGateway: IDatabaseGateway,
+) : ITopicManagementUseCase {
+    override suspend fun createTopic(topic: String): Boolean {
+        if (databaseGateway.isTopicAlreadyExists(topic)) throw InternalServerErrorException(ALREADY_EXISTS_TOPIC)
```

Comment:
Here if the resource already exist, you have to throw ResourceAlreadyExistException, with status code 409(Conflict)